js/packages/teams-ai/src/MessageExtensions.ts

```diff
@@ -540,6 +540,160 @@ export class MessageExtensions<TState extends TurnState> {
             });
         }
     }
+
+    /**
+     * Registers a handler that invokes the fetch of the configuration settings for a Message Extension.
+     @summary
+     * The `composeExtension/querySettingUrl` INVOKE activity does not contain a command ID, so only a single select item handler can be registered.
+     * @param {(context: TurnContext, state: TState) => Promise<void>} handler Function defined by the developer to call when the command is received.
+     * @param {TurnContext} handler.context Context for the current turn of conversation with the user.
+     * @param {TState} handler.state Current state of the turn.
+     * @returns {Application<TState>} The application for chaining purposes.
+     */
+    public queryUrlSetting(handler: (context: TurnContext, state: TState) => Promise<void>): Application<TState> {
+        // Define static route selector
+        const selector = (context: TurnContext) =>
+            Promise.resolve(
+                context?.activity?.type == ActivityTypes.Invoke && context?.activity.name === QUERY_SETTING_URL
+            );
+
+        // Add route
+        this._app.addRoute(
+            selector,
+            async (context, state) => {
+                try {
+                    // Call handler and then check to see if an invoke response has already been added
+                    await handler(context, state);
```

Comment:
The handler should return a `MessageExtensionResponse` and this should be as part of the "value" object in the activity that's being sent:
```js
     * Receives invoke activities with the name 'composeExtension/querySettingUrl'
     *
     * @param _context A context object for this turn.
     * @param _query The Messaging extension query.
     * @returns The Messaging Extension Action Response for the query.
     */
    protected handleTeamsMessagingExtensionConfigurationQuerySettingUrl(_context: TurnContext, _query: MessagingExtensionQuery): Promise<MessagingExtensionResponse>;
```